From d51d79900eb9fa455ff3b87da59ab3b4e492686e Mon Sep 17 00:00:00 2001
From: Neil Chen <Chen.NeilZX@inventec.com>
Date: Thu, 13 May 2021 03:01:02 +0000
Subject: [PATCH 33/34] - Modify RGMII TX Clock delay and TX driving strength

    Symtom/Reason:
        RGMII signal can't pass in EV test

    Root Cause:
        RGMII TX Clock delay isn't enough.
        Amplitude of TX data isn't enough

    Solution/Change:
        Add TX clock delay time.
        Add MAC# driving strength

    Entry Test:
	sysadmin@transformers:~# devmem 0x1e6e2350
	0x001043C8
	sysadmin@transformers:~# devmem 0x1e6e2458
	0x0000000F
---
 drivers/clk/clk-ast2600.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/clk/clk-ast2600.c b/drivers/clk/clk-ast2600.c
index fc98686eaf78..0648edfe0d24 100644
--- a/drivers/clk/clk-ast2600.c
+++ b/drivers/clk/clk-ast2600.c
@@ -41,7 +41,7 @@
 #define ASPEED_G6_DEF_MAC12_DELAY_1G	0x0041b410
 #define ASPEED_G6_DEF_MAC12_DELAY_100M	0x00417410
 #define ASPEED_G6_DEF_MAC12_DELAY_10M	0x00417410
-#define ASPEED_G6_DEF_MAC34_DELAY_1G	0x00104208
+#define ASPEED_G6_DEF_MAC34_DELAY_1G	0x001043C8
 #define ASPEED_G6_DEF_MAC34_DELAY_100M	0x00104208
 #define ASPEED_G6_DEF_MAC34_DELAY_10M	0x00104208
 
@@ -1122,7 +1122,7 @@ static void __init aspeed_g6_cc_init(struct device_node *np)
 		     ASPEED_G6_DEF_MAC34_DELAY_10M);
 
 	/* MAC3/4 default pad driving strength */
-	regmap_write(map, ASPEED_G6_MAC34_DRIVING_CTRL, 0x0000000a);
+	regmap_write(map, ASPEED_G6_MAC34_DRIVING_CTRL, 0x0000000f);
 	
 	/* RSA clock = HPLL/3 */
 	regmap_update_bits(map, ASPEED_G6_CLK_SELECTION1, BIT(19), BIT(19));	
-- 
2.33.0

